src/components/layout/minibag.js

```diff
@@ -17,52 +14,80 @@ class Minibag extends PureComponent {
+              </this.Link>
+            </div>
+          </div>
+        )
+      })
+
+      return cartData
+    }
+    
+    return null
   }
 
-  /**
-   * Renders the checkout link. Disabled if the cart is empty
-   * @param  {number} lineItemCount
-   * @return {string} - HTML markup for the component
-   */
-  renderCheckoutLink (lineItemCount) {
+  renderMiniBagDropdown(lineItemsCount, lineItems, total, shippingTotal) {
+    const miniBagTotal = total - shippingTotal
     return (
-      <div className='c-minibag__checkout'>
-        <this.Link
-          href={(lineItemCount > 0) ? '/checkout' : ''}
-          className={classNames('o-button o-button--primary', { 'o-button--disabled': (lineItemCount === 0) })}>
-            Checkout
-        </this.Link>
+      <div className='c-minibag__dropdown'>
+        <div className='c-minibag__dropdown-container'>
+          <section className='c-minibag__dropdown-header'>
+            <h1 className='c-minibag__dropdown-title'> Shopping Basket <a className='c-checkout-cart__amount'>({lineItemsCount})</a></h1>
+            <input id='minibag' type='checkbox' className='c-minibag__dropdown-checkbox' checked={this.props.miniBagDisplayed} readOnly />
+            <label htmlFor='minibag' className='c-minibag__dropdown-cross' onClick={this.props.toggleMiniBag} />
+          </section>
+          <div className='c-minibag__line-items-section'>
+            { this.renderLineItems(lineItems, lineItemsCount) }
+          </div>
+          <div className='c-minibag__dropdown-review'>
+            <span className='c-minibag__dropdown-review-total'>
+              <h4>Total:</h4>
+              <h4>&pound;{ decimalPrice(miniBagTotal) }</h4>
+            </span>
+            <div className='c-minibag__dropdown-buttons'>
+              <this.Link href='/cart' className='o-button o-button--sml o-button--primary c-minibag__dropdown-buttons--link'>
+                view shopping basket
+              </this.Link>
+              <this.Button label='continue shopping' className='o-button--sml c-minibag__dropdown-buttons--link' status='primary' onClick={() => this.props.toggleMiniBag} />
+            </div>
+          </div>
+        </div>
       </div>
     )
   }
 
   render () {
-    const { cart, className } = this.props
-    const lineItemCount = cart.line_items_count || 0
+    const { cart } = this.props
+    const lineItemsCount = cart.line_items_count || 0
+    const lineItems = cart.line_items
+    const miniBagDisplayed = cart.miniBagDisplayed || this.props.miniBagDisplayed
 
-    return (
-      <div className={classNames(className, 'c-header__minibag c-minibag')}>
-        { this.renderCartLink(lineItemCount) }
-        { this.renderCheckoutLink(lineItemCount) }
-      </div>
-    )
+    return <>
+        { (miniBagDisplayed && lineItemsCount > 0) && this.renderMiniBagDropdown(lineItemsCount, lineItems, cart.total, cart.shipping_total) }
+    </>
```

Comment:
Do we still need this fragment? If it's here to handle the case when the conditional above is false then we could potentially return `null` instead IIRC.